File: objects/cpu.cxx
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: http://www.mikekohn.net/
 * License: GPL
 *
 * Copyright 2014 by Michael Kohn
 *
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdint.h>

#include "JavaClass.h"
#include "compile.h"
#include "cpu.h"

#define CHECK_FUNC(funct) \
  if (strncmp(#funct, function, sizeof(#funct)-1) == 0) \
  { \
    return cpu_##funct(java_class, generator); \
  }

int cpu_setClock16(JavaClass *java_class, Generator *generator)
{
  return generator->cpu_setClock16();
}

int cpu_nop(JavaClass *java_class, Generator *generator)
{
  return generator->cpu_nop();
}

int cpu(JavaClass *java_class, Generator *generator, char *function)
{
  CHECK_FUNC(setClock16)
  CHECK_FUNC(nop)

  return -1;
}

File: java/net/mikekohn/java_grinder/CPU.java
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: https://www.mikekohn.net/
 * License: GPL
 *
 * Copyright 2014-2022 by Michael Kohn
 *
 */

package net.mikekohn.java_grinder;

// This class is a bit awkward. Not sure how to make it better right now. 

public class CPU
{
  private CPU()
  {
  }

  /** Set clock speed to 1MHz (internal crystal? DCO?) */
  public static void setClock1() { }

  /** Set clock speed to 2MHz (internal crystal? DCO?) */
  public static void setClock2() { }

  /** Set clock speed to 4MHz (internal crystal? DCO?) */
  public static void setClock4() { }

  /** Set clock speed to 8MHz (internal crystal? DCO?) */
  public static void setClock8() { }

  /** Set clock speed to 16MHz (internal crystal? DCO?) */
  public static void setClock16() { }

  /** Set clock speed to 25MHz (internal crystal? DCO?) */
  public static void setClock25() { }

  /** Set clock speed to XT2 (external crystal) */
  public static void setClockExternal2() { }

  /** Insert a CPU level NOP */
  public static void nop() { }

  /** Insert inline assembly */
  public static void asm(String code) { }

  /** Get cycle counter from CPU (if available) { } */
  public static int getCycleCount() { return 0; }
}

